crates/kubelet/src/volume/projected.rs

```diff
@@ -0,0 +1,250 @@
+use std::path::Path;
+
+use either::Either;
+use k8s_openapi::api::authentication::v1::{BoundObjectReference, TokenRequest, TokenRequestSpec};
+use k8s_openapi::api::core::v1::{
+    ConfigMapVolumeSource, DownwardAPIVolumeSource, Pod as KubePod, SecretVolumeSource,
+    Volume as KubeVolume, VolumeProjection,
+};
+use k8s_openapi::Resource;
+use tracing::warn;
+
+use super::*;
+
+/// A type that can manage a Secret volume with mounting and unmounting support
+pub struct ProjectedVolume {
+    vol_name: String,
+    volumes: Vec<super::VolumeRef>,
+    service_accounts: Vec<ServiceAccountSource>,
+    mounted_path: Option<PathBuf>,
+}
+
+struct ServiceAccountSource {
+    file_name: String,
+    service_account_name: String,
+    namespace: String,
+    client: kube::Client,
+    audience: String,
+    expiration_time: i64,
+    pod_name: String,
+    pod_uid: String,
+}
+
+impl ServiceAccountSource {
+    async fn mount_at(&mut self, path: impl AsRef<Path>) -> anyhow::Result<()> {
+        // As far as I can tell, this is the only way to access the token subresource on service accounts
```

Comment:
Yeah, this comment from kubeflow seems to support that statement (short of storing this information in another service account): https://github.com/kubeflow/pipelines/issues/5138#issuecomment-845846555